### PR TITLE
Add mev-boost support

### PR DESCRIPTION
**Pull request by nflaig**
Adds support for mev-boost

@dapplion

### CODE REVIEW

File: validator/entrypoint.sh [hunk 1]
```diff
@@ -1,5 +1,17 @@
 #!/bin/sh
 
+# MEVBOOST: https://chainsafe.github.io/lodestar/usage/mev-integration/
+if [ -n "$_DAPPNODE_GLOBAL_MEVBOOST_MAINNET" ] && [ "$_DAPPNODE_GLOBAL_MEVBOOST_MAINNET" == "true" ]; then
+  echo "MEVBOOST is enabled"
```
Comment (alexpeterson91) [inline validator/entrypoint.sh]:
```suggestion
  echo "MEV-Boost is enabled"
```

File: validator/entrypoint.sh [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+  MEVBOOST_URL="http://mev-boost.mev-boost.dappnode:18550"
+  if curl --retry 5 --retry-delay 5 --retry-all-errors "${MEVBOOST_URL}"; then
+    EXTRA_OPTS="--builder ${EXTRA_OPTS}"
+  else
+    echo "MEVBOOST is enabled but ${MEVBOOST_URL} is not reachable"
```
Comment (alexpeterson91) [inline validator/entrypoint.sh]:
```suggestion
    echo "MEV-Boost is enabled but the Mainnet MEV-Boost package at ${MEVBOOST_URL} is not reachable"
```

File: validator/entrypoint.sh [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+    curl -X POST -G 'http://my.dappnode/notification-send' --data-urlencode 'type=danger' --data-urlencode title="${MEVBOOST_URL} is not available" --data-urlencode 'body=Make sure the mevboost is available and running'
```
Comment (alexpeterson91) [inline validator/entrypoint.sh]:
```suggestion
    curl -X POST -G 'http://my.dappnode/notification-send' --data-urlencode 'type=danger' --data-urlencode title="${MEVBOOST_URL} can not be reached" --data-urlencode 'body=Make sure the Mainnet MEV-Boost DNP is available and running'
```

File: dappnode_package.json [hunk 4]
```diff
@@ -35,5 +35,15 @@
   },
   "warnings": {
     "onRemove": "Make sure your web3signer does not have this client selected! (Packages > web3signer > config > client)"
```
Comment (alexpeterson91) [inline dappnode_package.json]:
```suggestion
    "onRemove": "Make sure your StakersUI does not have this client selected! Double check in the Stakers Tab in the left NavBar"
```

File: beacon-chain/entrypoint.sh [hunk 5]
```diff
@@ -1,6 +1,13 @@
 # Concatenate EXTRA_OPTS string
 [[ -n "$CHECKPOINT_SYNC_URL" ]] && EXTRA_OPTS="${EXTRA_OPTS} --checkpointSyncUrl=${CHECKPOINT_SYNC_URL}"
 
+# MEVBOOST: https://chainsafe.github.io/lodestar/usage/mev-integration/
+if [ -n "$_DAPPNODE_GLOBAL_MEVBOOST_MAINNET" ] && [ "$_DAPPNODE_GLOBAL_MEVBOOST_MAINNET" == "true" ]; then
+    echo "MEVBOOST is enabled"
```
Comment (alexpeterson91) [inline beacon-chain/entrypoint.sh]:
```suggestion
    echo "MEV-Boost is enabled"
```

**Review by alexpeterson91 — APPROVED** [review]

**Comment by nflaig** [comment]
thx for the update @alexpeterson91, I think we should soon be able to have official release of the Lodestar prater package and once that is fully tested, I will double check the mainnet package and apply any inconsistencies/missing features.

What do you think we should do about the chiado and gnosis package? It should be quite easy to update those or rather in case of gnosis create it

**Comment by alexpeterson91** [comment]
> thx for the update @alexpeterson91, I think we should soon be able to have official release of the Lodestar prater package and once that is fully tested, I will double check the mainnet package and apply any inconsistencies/missing features.
> 
> 
> 
> What do you think we should do about the chiado and gnosis package? It should be quite easy to update those or rather in case of gnosis create it

Sounds great!  I was going to ask you if the latest upstream version is good to merge. As for the gnosis package I think it's best to get the prater release done and then just make a new repo from a template of the prater repo then just go around and make the small changes needed to adapt for another chain in this case gnosis mainnet. 
And for chiado I'm not sure last @dapplion and I discussed it which was a while ago now was that we weren't gonna put the effort into chiado for any clients at least now while the validator set is still limited and controlled by the gnosis team. So until they open chiado for public validation again i don't think we should be too worried about it. We'd need to spin up a new stakersUI tab and update nethermind Chiado and the other CL clients that work with chiado too lodestar just made its first official appearance on dappnode during the first round of chiado which didn't last very long and had very limited participation (Lion had given us a few hundred chiado keys and I setup a ticket request for batches of 10 chiado keys, we only got requests from like 16 people total for chiado keys and only like half even used them.) Also this was before we started using the stakersUI and Global_ENVs for all chains etc.

**Comment by nflaig** [comment]
> I think it's best to get the prater release done and then just make a new repo from a template of the prater repo then just go around and make the small changes needed to adapt for another chain in this case gnosis mainnet.

sounds good to me, the changes are really minimal, I think we just need to set the network parameter to `gnosis`. Upstream dependencies are only [DAppNodePackage-web3signer-gnosis](https://github.com/dappnode/DAppNodePackage-web3signer-gnosis) and [DAppNodePackage-nethermind-xdai](https://github.com/dappnode/DAppNodePackage-nethermind-xdai) as far as I've seen.

> put the effort into chiado for any clients at least now while the validator set is still limited and controlled by the gnosis team

would agree with that, there are too many missing gaps to support chiado which are not specific to the Lodestar package

**Comment by nflaig** [comment]
@alexpeterson91 do you know what the dappnode policy for doppelganger protection is? Does not seem to be consistent between CLs, in case of the Lodestar package we have it enabled now on mainnet but disabled on prater.

**Pull request merged**